frank-flow/src/frontend/src/app/flow/adapter-selector/adapter-selector.component.ts

```diff
@@ -0,0 +1,48 @@
+import { Component, OnDestroy, OnInit, ViewChild } from '@angular/core';
+import { File } from '../../shared/models/file.model';
+import { CurrentFileService } from '../../shared/services/current-file.service';
+import { Subscription } from 'rxjs';
+
+@Component({
+  selector: 'app-adapter-selector',
+  templateUrl: './adapter-selector.component.html',
+  styleUrls: ['./adapter-selector.component.scss'],
+})
+export class AdapterSelectorComponent implements OnInit, OnDestroy {
+  public currentFile!: File
+  public adapterNames?: string[]
+  public currentAdapterName = ""
+
+  private currentFileSubscription!: Subscription
+
+  constructor(private currentFileService: CurrentFileService) {
+  }
+
+  ngOnInit(): void {
+    this.subscribeToCurrentFile();
+  }
+
+  ngOnDestroy(): void {
+    this.currentFileSubscription.unsubscribe();
+  }
+
+  subscribeToCurrentFile(): void {
+    this.currentFileSubscription =
+      this.currentFileService.currentFileObservable.subscribe({
+        next: (currentFile: File): void => {
```

Comment:
why use `subscribe({next: ((...) => void)})` when you dont use `error: ((error) => void)` or `complete: (() => {})`?
You could just use the normal arrow function way of calling subscribe: `subscribe((...) => void)`

Reply:
In the past we used this as convention, everything needed to be as explicit as possible. This would also give the added benefit of having less changes (in git) when `error` or `complete` gets added in the future.

Conventions like these need to be discussed in the whole organization.

Reply:
Okay this convention needs a way better definition.
Don't get me wrong, I get what you mean but now we don't have a real boundary of how far "everything to be as explicit as possible" has to go.

For example, I bet you like lambda's or arrow functions, that is not possible to be used if we follow the literal definition of this convention since a full function or class implementation is "as explicit as possible".
We should make a wiki or file somewhere where we put these kinds of conventions and their boundaries.
Let's talk about it tomorrow if my example isn't clear, I think we can easily find a good middle ground

Reply:
I totally agree. I used the term "as explicit as possible" as summery, but in general it's true. Things like this got discussed in person and since the team was just two people that reviewed each other, this was good enough. Now we are two years and 3 team members later and it's obvious how bad of a decision it was to not document it.

It might be better to discuss this for the Frank!Console first, since it has the right linting already. There is also a lot of code that still needs to be refactored, it can than be matched to the new conventions.

Reply:
This sounds like good idea, I'll make an issue in the Frank!Console for it.
Sadly it doesn't look like there is an org wide wiki or discussions page so well have to figure out where to document such conventions

Reply:
Issue can be found here: https://github.com/frankframework/frankframework/issues/6411